deployment/cloud-config.yml

```diff
@@ -0,0 +1,27 @@
+#cloud-config
+mounts:
+- [ /dev/nvme0n1p1, /mnt/data/ ]
+
+packages:
+# These dependencies are nessecary to build Onramp Python dependencies
+- build-essential
+- python3-dev
+# For isolating Onramp build dependencies
+- python3-venv
+
+runcmd:
+# Install OSMExpress
+- curl -L https://github.com/protomaps/OSMExpress/releases/download/0.2.0/osmexpress-0.2.0-Linux.tgz | tar -xzC /usr/local/bin/ osmx
+# Clone Onramp
+- git clone https://github.com/azavea/onramp.git /opt/onramp 
+- sudo chown -R ubuntu:ubuntu /opt/onramp/
+- cd /opt/onramp/app
+# Install Python dependencies
+- python3 -m venv ./venv
+- sudo chown -R ubuntu:ubuntu ./venv
+- ./venv/bin/pip install wheel
+- ./venv/bin/pip install -r requirements.txt
+# Initial invocation of osmx-update
+- su - ubuntu -c "/opt/onramp/app/venv/bin/python /opt/onramp/app/osmx-update /mnt/data/planet.osmx https://planet.openstreetmap.org/replication/minute/ 2>&1 | /usr/bin/logger -t osmx-update"
```

Comment:
Currently this command does not generate augmented diffs, but once we've battle tested it a bit in the client environment, we'll want to switch on the `--augmented-diffs` flag of the `osmx-update` command. I could envision a world where a user just wants quick deployment of osmexpress without the augmented diffs (coming in potentially another of our client projects) so is it possible to parameterize cloud init such that a user could choose whether to enable augmented diff generation when they run the `aws ec2 run-instances` command?

Reply:
There isn't a way to parameterize cloud init without threading it through another tool like Terraform or CloudFormation.

We could do something like this.

```bash
sed -e "s/%%GIT_COMMIT%%/master/" \
    -e "s/%%OSMX_UPDATE_OPTS%%//" \
    "cloud-config.yml.template" > cloud-config.yml
```

Where `GIT_COMMIT` parameterizes the ref to checkout, and `OSMX_UPDATE_OPTS` would allow us to pass additional parameters to the `osmx-update` commands.

What are your thoughts on the ergonomics of this?

Reply:
That seems acceptable for now given the limited usage and number of variables we would need to parameterize. We'll just want to watch for if/when it becomes unwieldy to explore other options.